Lots of lint errors/warnings

```
jsonpickle bugfix/lint-awareness $ flake8
./setup.py:25:81: E501 line too long (86 > 80 characters)
./setup.py:26:81: E501 line too long (84 > 80 characters)
./tests/benchmark.py:35:16: E999 SyntaxError: invalid syntax
./tests/document_test.py:102:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/numpy_test.py:116:81: E501 line too long (81 > 80 characters)
./tests/jsonpickle_test.py:1378:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/sqlalchemy_test.py:98:17: E128 continuation line under-indented for visual indent
./tests/sqlalchemy_test.py:99:17: E128 continuation line under-indented for visual indent
./tests/sqlalchemy_test.py:119:1: E302 expected 2 blank lines, found 1
./tests/util_test.py:241:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/backend_test.py:139:13: W605 invalid escape sequence '\/'
./tests/backend_test.py:166:13: W605 invalid escape sequence '\/'
./tests/object_test.py:747:9: E731 do not assign a lambda expression, use a def
./tests/object_test.py:795:20: E225 missing whitespace around operator
./tests/object_test.py:808:16: E225 missing whitespace around operator
./tests/object_test.py:858:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/datetime_test.py:37:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/datetime_test.py:262:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/runtests.py:16:1: E402 module level import not at top of file
./tests/runtests.py:18:1: E402 module level import not at top of file
./tests/runtests.py:19:1: E402 module level import not at top of file
./tests/runtests.py:20:1: E402 module level import not at top of file
./tests/runtests.py:21:1: E402 module level import not at top of file
./tests/runtests.py:22:1: E402 module level import not at top of file
./tests/runtests.py:23:1: E402 module level import not at top of file
./tests/runtests.py:24:1: E402 module level import not at top of file
./tests/runtests.py:25:1: E402 module level import not at top of file
./tests/runtests.py:26:1: E402 module level import not at top of file
./tests/runtests.py:27:1: E402 module level import not at top of file
./tests/runtests.py:28:1: E402 module level import not at top of file
./tests/handler_test.py:58:1: E303 too many blank lines (3)
./contrib/pyqt-reduce-handler.py:20:20: F821 undefined name 'unicode'
./contrib/pyqt-reduce-handler.py:22:81: E501 line too long (83 > 80 characters)
./contrib/pyqt-reduce-handler.py:38:1: E305 expected 2 blank lines after class or function definition, found 1
./jsonpickle/unpickler.py:4:81: E501 line too long (87 > 80 characters)
./jsonpickle/unpickler.py:18:1: F401 '.backend.JSONBackend' imported but unused
./jsonpickle/unpickler.py:201:13: E731 do not assign a lambda expression, use a def
./jsonpickle/unpickler.py:223:9: F841 local variable 'reduce_tuple' is assigned to but never used
./jsonpickle/unpickler.py:223:81: E501 line too long (92 > 80 characters)
./jsonpickle/unpickler.py:252:21: E722 do not use bare 'except'
./jsonpickle/unpickler.py:347:81: E501 line too long (83 > 80 characters)
./jsonpickle/unpickler.py:349:81: E501 line too long (82 > 80 characters)
./jsonpickle/unpickler.py:366:17: E722 do not use bare 'except'
./jsonpickle/unpickler.py:480:21: E127 continuation line over-indented for visual indent
./jsonpickle/unpickler.py:481:25: E127 continuation line over-indented for visual indent
./jsonpickle/unpickler.py:524:13: E731 do not assign a lambda expression, use a def
./jsonpickle/util.py:66:81: E501 line too long (84 > 80 characters)
./jsonpickle/util.py:67:81: E501 line too long (89 > 80 characters)
./jsonpickle/util.py:69:81: E501 line too long (89 > 80 characters)
./jsonpickle/util.py:345:5: E722 do not use bare 'except'
./jsonpickle/util.py:351:1: E302 expected 2 blank lines, found 1
./jsonpickle/util.py:353:81: E501 line too long (83 > 80 characters)
./jsonpickle/util.py:360:17: E127 continuation line over-indented for visual indent
./jsonpickle/util.py:361:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:362:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:363:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:364:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:365:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:366:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:367:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:368:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:369:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:370:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:371:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:372:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:373:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:374:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:375:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:376:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:377:17: E128 continuation line under-indented for visual indent
./jsonpickle/util.py:378:17: E124 closing bracket does not match visual indentation
./jsonpickle/util.py:394:81: E501 line too long (81 > 80 characters)
./jsonpickle/util.py:412:10: E111 indentation is not a multiple of four
./jsonpickle/util.py:422:72: W291 trailing whitespace
./jsonpickle/util.py:432:81: E501 line too long (92 > 80 characters)
./jsonpickle/util.py:438:10: E111 indentation is not a multiple of four
./jsonpickle/util.py:439:10: E111 indentation is not a multiple of four
./jsonpickle/util.py:439:13: E714 test for object identity should be 'is not'
./jsonpickle/util.py:440:14: E111 indentation is not a multiple of four
./jsonpickle/util.py:444:12: E714 test for object identity should be 'is not'
./jsonpickle/util.py:445:14: E111 indentation is not a multiple of four
./jsonpickle/handlers.py:7:81: E501 line too long (86 > 80 characters)
./jsonpickle/handlers.py:24:1: F401 'collections' imported but unused
./jsonpickle/handlers.py:27:1: F401 'decimal' imported but unused
./jsonpickle/handlers.py:29:1: F401 'sys' imported but unused
./jsonpickle/handlers.py:30:1: F401 'time' imported but unused
./jsonpickle/handlers.py:49:81: E501 line too long (91 > 80 characters)
./jsonpickle/handlers.py:50:81: E501 line too long (90 > 80 characters)
./jsonpickle/handlers.py:53:81: E501 line too long (89 > 80 characters)
./jsonpickle/handlers.py:63:81: E501 line too long (91 > 80 characters)
./jsonpickle/handlers.py:64:81: E501 line too long (90 > 80 characters)
./jsonpickle/handlers.py:66:81: E501 line too long (90 > 80 characters)
./jsonpickle/handlers.py:82:81: E501 line too long (81 > 80 characters)
./jsonpickle/handlers.py:217:1: E305 expected 2 blank lines after class or function definition, found 1
./jsonpickle/handlers.py:231:81: E501 line too long (81 > 80 characters)
./jsonpickle/handlers.py:244:1: E305 expected 2 blank lines after class or function definition, found 1
./jsonpickle/__init__.py:58:1: F401 '.pickler' imported but unused
./jsonpickle/__init__.py:59:1: F401 '.unpickler' imported but unused
./jsonpickle/__init__.py:60:1: F401 '.backend.JSONBackend' imported but unused
./jsonpickle/__init__.py:61:1: F401 '.version.__version__' imported but unused
./jsonpickle/__init__.py:63:1: F401 '.handlers.register' imported but unused
./jsonpickle/__init__.py:64:1: F401 '.handlers.unregister' imported but unused
./jsonpickle/__init__.py:65:1: F401 '.pickler.Pickler' imported but unused
./jsonpickle/__init__.py:66:1: F401 '.unpickler.Unpickler' imported but unused
./jsonpickle/pickler.py:4:81: E501 line too long (87 > 80 characters)
./jsonpickle/pickler.py:18:1: F401 '.backend.JSONBackend' imported but unused
./jsonpickle/pickler.py:209:36: F821 undefined name 'file'
./jsonpickle/pickler.py:270:44: F821 undefined name 'file'
./jsonpickle/pickler.py:277:13: E722 do not use bare 'except'
./jsonpickle/pickler.py:364:29: E127 continuation line over-indented for visual indent
./jsonpickle/pickler.py:365:29: E127 continuation line over-indented for visual indent
./jsonpickle/pickler.py:420:81: E501 line too long (87 > 80 characters)
./jsonpickle/pickler.py:531:17: E722 do not use bare 'except'
./jsonpickle/ext/numpy.py:20:1: E302 expected 2 blank lines, found 1
./jsonpickle/ext/numpy.py:83:81: E501 line too long (90 > 80 characters)
./jsonpickle/ext/numpy.py:103:81: E501 line too long (89 > 80 characters)
./jsonpickle/ext/numpy.py:107:81: E501 line too long (101 > 80 characters)
./jsonpickle/ext/numpy.py:113:81: E501 line too long (82 > 80 characters)
./jsonpickle/ext/numpy.py:140:48: E261 at least two spaces before inline comment
./jsonpickle/ext/numpy.py:140:81: E501 line too long (121 > 80 characters)
./jsonpickle/ext/numpy.py:142:49: E261 at least two spaces before inline comment
./jsonpickle/ext/numpy.py:175:21: E261 at least two spaces before inline comment
./jsonpickle/ext/numpy.py:189:81: E501 line too long (89 > 80 characters)
./jsonpickle/ext/numpy.py:190:81: E501 line too long (114 > 80 characters)
./jsonpickle/ext/numpy.py:191:81: E501 line too long (118 > 80 characters)
./jsonpickle/ext/numpy.py:193:81: E501 line too long (97 > 80 characters)
./jsonpickle/ext/numpy.py:194:81: E501 line too long (87 > 80 characters)
./jsonpickle/ext/numpy.py:195:81: E501 line too long (96 > 80 characters)
./jsonpickle/ext/numpy.py:200:81: E501 line too long (104 > 80 characters)
./jsonpickle/ext/numpy.py:202:81: E501 line too long (82 > 80 characters)
./jsonpickle/ext/numpy.py:208:81: E501 line too long (81 > 80 characters)
./jsonpickle/ext/numpy.py:217:81: E501 line too long (82 > 80 characters)
./jsonpickle/ext/numpy.py:236:81: E501 line too long (86 > 80 characters)
./jsonpickle/ext/numpy.py:241:81: E501 line too long (101 > 80 characters)
./jsonpickle/ext/numpy.py:246:81: E501 line too long (104 > 80 characters)
./jsonpickle/ext/numpy.py:247:81: E501 line too long (84 > 80 characters)
./jsonpickle/ext/numpy.py:250:81: E501 line too long (101 > 80 characters)
./jsonpickle/ext/numpy.py:252:81: E501 line too long (81 > 80 characters)
./jsonpickle/ext/numpy.py:270:81: E501 line too long (88 > 80 characters)
```

--- a/contrib/pyqt-reduce-handler.py
+++ b/contrib/pyqt-reduce-handler.py
@@ -11,15 +11,18 @@ from PyQt4 import QtCore
 import jsonpickle
 from jsonpickle import handlers
 
+text_type = eval('unicode') if str is bytes else str
+
 
 class QReduceHandler(handlers.BaseHandler):
 
     def flatten(self, obj, data):
         pickler = self.context
         if not pickler.unpicklable:
-            return unicode(obj)
+            return text_type(obj)
         flatten = pickler.flatten
-        data['__reduce__'] = [flatten(i, reset=False) for i in obj.__reduce__()[1]]
+        data['__reduce__'] = [
+            flatten(i, reset=False) for i in obj.__reduce__()[1]]
         return data
 
     def restore(self, data):
@@ -34,6 +37,7 @@ class QReduceHandler(handlers.BaseHandler):
         factory = getattr(module, classname)
 
         return factory(*args)
+
 
 handlers.register(QtCore.QPointF, QReduceHandler)
 

--- a/jsonpickle/__init__.py
+++ b/jsonpickle/__init__.py
@@ -55,17 +55,16 @@ added to JSON::
 """
 from __future__ import absolute_import, division, unicode_literals
 
-from . import pickler
-from . import unpickler
-from .backend import JSONBackend
-from .version import __version__
+import importlib
+
 from .backend import json
-from .handlers import register  # side-effect: registers built-in handlers
-from .handlers import unregister
-from .pickler import Pickler, encode
-from .unpickler import Unpickler, decode
+from .pickler import encode
+from .unpickler import decode
 
 __all__ = ('encode', 'decode')
+
+# register built-in handlers
+importlib.import_module('.handlers', package=__name__)
 
 # Export specific JSONPluginMgr methods into the jsonpickle namespace
 set_preferred_backend = json.set_preferred_backend

--- a/jsonpickle/__init__.py
+++ b/jsonpickle/__init__.py
@@ -74,6 +74,6 @@ load_backend = json.load_backend
 remove_backend = json.remove_backend
 enable_fallthrough = json.enable_fallthrough
 
-# json.load(),loads(), dump(), dumps() compatibility
+# json.load(), loads(), dump(), dumps() compatibility
 dumps = encode
 loads = decode

--- a/jsonpickle/__init__.py
+++ b/jsonpickle/__init__.py
@@ -55,8 +55,6 @@ added to JSON::
 """
 from __future__ import absolute_import, division, unicode_literals
 
-import importlib
-
 from .backend import json
 from .pickler import encode
 from .unpickler import decode
@@ -72,7 +70,7 @@ from .unpickler import Unpickler  # noqa: F401
 __all__ = ('encode', 'decode')
 
 # register built-in handlers
-importlib.import_module('.handlers', package=__name__)
+__import__('jsonpickle.handlers', level=0)
 
 # Export specific JSONPluginMgr methods into the jsonpickle namespace
 set_preferred_backend = json.set_preferred_backend

--- a/jsonpickle/__init__.py
+++ b/jsonpickle/__init__.py
@@ -61,6 +61,14 @@ from .backend import json
 from .pickler import encode
 from .unpickler import decode
 
+# Export other names not in __all__
+from .backend import JSONBackend  # noqa: F401
+from .version import __version__  # noqa: F401
+from .handlers import register  # noqa: F401
+from .handlers import unregister  # noqa: F401
+from .pickler import Pickler  # noqa: F401
+from .unpickler import Unpickler  # noqa: F401
+
 __all__ = ('encode', 'decode')
 
 # register built-in handlers

--- a/jsonpickle/ext/numpy.py
+++ b/jsonpickle/ext/numpy.py
@@ -17,6 +17,7 @@ __all__ = ['register_handlers', 'unregister_handlers']
 
 native_byteorder = '<' if sys.byteorder == 'little' else '>'
 
+
 def get_byteorder(arr):
     """translate equals sign to native order"""
     byteorder = arr.dtype.byteorder
@@ -80,7 +81,8 @@ class NumpyNDArrayHandler(NumpyBaseHandler):
         self.flatten_flags(obj, data)
         data['values'] = self.context.flatten(obj.tolist(), reset=False)
         if 0 in obj.shape:
-            # add shape information explicitly as it cannot be inferred from an empty list
+            # add shape information explicitly as it cannot be
+            # inferred from an empty list
             data['shape'] = obj.shape
         return data
 
@@ -100,17 +102,20 @@ class NumpyNDArrayHandler(NumpyBaseHandler):
 
 
 class NumpyNDArrayHandlerBinary(NumpyNDArrayHandler):
-    """stores arrays with size greater than 'size_treshold' as (optionally) compressed base64
+    """stores arrays with size greater than 'size_treshold' as
+    (optionally) compressed base64
 
     Notes
     -----
-    This would be easier to implement using np.save/np.load, but that would be less language-agnostic
+    This would be easier to implement using np.save/np.load, but
+    that would be less language-agnostic
     """
 
     def __init__(self, size_treshold=16, compression=zlib):
         """
         :param size_treshold: nonnegative int or None
-            valid values for 'size_treshold' are all nonnegative integers and None
+            valid values for 'size_treshold' are all nonnegative
+            integers and None
             if size_treshold is None, values are always stored as nested lists
         :param compression: a compression module or None
             valid values for 'compression' are {zlib, bz2, None}
@@ -137,9 +142,12 @@ class NumpyNDArrayHandlerBinary(NumpyNDArrayHandler):
         else:
             # encode as binary
             if hasattr(obj, 'tobytes'):
-                buffer = obj.tobytes(order='a') # numpy docstring is lacking as of 1.11.2, but this is the option we need
+                # numpy docstring is lacking as of 1.11.2,
+                # but this is the option we need
+                buffer = obj.tobytes(order='a')
             else:
-                buffer = obj.tostring(order='a') # numpy < 1.9 compatibility
+                # numpy < 1.9 compatibility
+                buffer = obj.tostring(order='a')
             if self.compression:
                 buffer = self.compression.compress(buffer)
             data['values'] = b64encode(buffer)
@@ -172,7 +180,7 @@ class NumpyNDArrayHandlerBinary(NumpyNDArrayHandler):
                 dtype=self.restore_dtype(data),
                 shape=data.get('shape'),
                 order=data.get('order', 'C')
-            ).copy() # make a copy, to force the result to own the data
+            ).copy()  # make a copy, to force the result to own the data
             self.restore_byteorder(data, arr)
             self.restore_flags(data, arr)
 
@@ -186,26 +194,35 @@ class NumpyNDArrayHandlerView(NumpyNDArrayHandlerBinary):
     -----
     The current implementation has some restrictions.
 
-    'base' arrays, or arrays which are viewed by other arrays, must be f-or-c-contiguous.
-    This is not such a large restriction in practice, because all numpy array creation is c-contiguous by default.
-    Relaxing this restriction would be nice though; especially if it can be done without bloating the design too much.
+    'base' arrays, or arrays which are viewed by other arrays,
+    must be f-or-c-contiguous.
+    This is not such a large restriction in practice, because all
+    numpy array creation is c-contiguous by default.
+    Relaxing this restriction would be nice though; especially if
+    it can be done without bloating the design too much.
 
-    Furthermore, ndarrays which are views of array-like objects implementing __array_interface__,
-    but which are not themselves nd-arrays, are deepcopied with a warning (by default),
-    as we cannot guarantee whatever custom logic such classes implement is correctly reproduced.
+    Furthermore, ndarrays which are views of array-like objects
+    implementing __array_interface__,
+    but which are not themselves nd-arrays, are deepcopied with
+    a warning (by default),
+    as we cannot guarantee whatever custom logic such classes
+    implement is correctly reproduced.
     """
     def __init__(self, mode='warn', size_treshold=16, compression=zlib):
         """
         :param mode: {'warn', 'raise', 'ignore'}
-            How to react when encountering array-like objects whos references we cannot safely serialize
+            How to react when encountering array-like objects whos
+            references we cannot safely serialize
         :param size_treshold: nonnegative int or None
-            valid values for 'size_treshold' are all nonnegative integers and None
+            valid values for 'size_treshold' are all nonnegative
+            integers and None
             if size_treshold is None, values are always stored as nested lists
         :param compression: a compression module or None
             valid values for 'compression' are {zlib, bz2, None}
             if compresion is None, no compression is applied
         """
-        super(NumpyNDArrayHandlerView, self).__init__(size_treshold, compression)
+        super(NumpyNDArrayHandlerView, self).__init__(
+            size_treshold, compression)
         self.mode = mode
 
     def flatten(self, obj, data):
@@ -214,7 +231,8 @@ class NumpyNDArrayHandlerView(NumpyNDArrayHandlerBinary):
         if base is None and obj.flags.forc:
             # store by value
             data = super(NumpyNDArrayHandlerView, self).flatten(obj, data)
-            # ensure that views on arrays stored as text are interpreted correctly
+            # ensure that views on arrays stored as text
+            # are interpreted correctly
             if not obj.flags.c_contiguous:
                 data['order'] = 'F'
         elif isinstance(base, np.ndarray) and base.flags.forc:
@@ -233,23 +251,33 @@ class NumpyNDArrayHandlerView(NumpyNDArrayHandlerBinary):
             self.flatten_flags(obj, data)
 
             if get_byteorder(obj) != '|':
-                byteorder = 'S' if get_byteorder(obj) != get_byteorder(base) else None
+                byteorder = (
+                    'S' if get_byteorder(obj) != get_byteorder(base) else None)
                 if byteorder:
                     data['byteorder'] = byteorder
 
             if self.size_treshold >= obj.size:
-                # not used in restore since base is present, but include values for human-readability
+                # not used in restore since base is present, but
+                # include values for human-readability
                 super(NumpyNDArrayHandlerBinary, self).flatten(obj, data)
         else:
             # store a deepcopy or fail
             if self.mode == 'warn':
-                msg = "ndarray is defined by reference to an object we do not know how to serialize. " \
-                      "A deep copy is serialized instead, breaking memory aliasing."
+                msg = (
+                    "ndarray is defined by reference to an object "
+                    "we do not know how to serialize. "
+                    "A deep copy is serialized instead, breaking "
+                    "memory aliasing."
+                )
                 warnings.warn(msg)
             elif self.mode == 'raise':
-                msg = "ndarray is defined by reference to an object we do not know how to serialize."
+                msg = (
+                    "ndarray is defined by reference to an object we do "
+                    "not know how to serialize."
+                )
                 raise ValueError(msg)
-            data = super(NumpyNDArrayHandlerView, self).flatten(obj.copy(), data)
+            data = super(NumpyNDArrayHandlerView, self) \
+                .flatten(obj.copy(), data)
 
         return data
 
@@ -267,7 +295,8 @@ class NumpyNDArrayHandlerView(NumpyNDArrayHandlerBinary):
 
             arr = np.ndarray(
                 buffer=base.data,
-                dtype=self.restore_dtype(data).newbyteorder(data.get('byteorder', '|')),
+                dtype=self.restore_dtype(data).newbyteorder(
+                    data.get('byteorder', '|')),
                 shape=data.get('shape'),
                 offset=data.get('offset', 0),
                 strides=data.get('strides', None)

--- a/jsonpickle/handlers.py
+++ b/jsonpickle/handlers.py
@@ -4,7 +4,8 @@ Custom handlers may be created to handle other objects. Each custom handler
 must derive from :class:`jsonpickle.handlers.BaseHandler` and
 implement ``flatten`` and ``restore``.
 
-A handler can be bound to other types by calling :func:`jsonpickle.handlers.register`.
+A handler can be bound to other types by calling
+:func:`jsonpickle.handlers.register`.
 
 :class:`jsonpickle.customhandlers.SimpleReduceHandler` is suitable for handling
 objects that implement the reduce protocol::
@@ -21,13 +22,9 @@ objects that implement the reduce protocol::
 
 """
 from __future__ import absolute_import, division, unicode_literals
-import collections
 import copy
 import datetime
-import decimal
 import re
-import sys
-import time
 import uuid
 
 from . import util
@@ -46,11 +43,14 @@ class Registry(object):
         :param cls_or_name: the type or its fully qualified name
         :param default: default value, if a matching handler is not found
 
-        Looks up a handler by type reference or its fully qualified name. If a direct match
-        is not found, the search is performed over all handlers registered with base=True.
+        Looks up a handler by type reference or its fully
+        qualified name. If a direct match
+        is not found, the search is performed over all
+        handlers registered with base=True.
         """
         handler = self._handlers.get(cls_or_name)
-        if handler is None and util.is_type(cls_or_name):  # attempt to find a base class
+        # attempt to find a base class
+        if handler is None and util.is_type(cls_or_name):
             for cls, base_handler in self._base_handlers.items():
                 if issubclass(cls_or_name, cls):
                     return base_handler
@@ -60,10 +60,13 @@ class Registry(object):
         """Register the a custom handler for a class
 
         :param cls: The custom object class to handle
-        :param handler: The custom handler class (if None, a decorator wrapper is returned)
-        :param base: Indicates whether the handler should be registered for all subclasses
+        :param handler: The custom handler class (if
+            None, a decorator wrapper is returned)
+        :param base: Indicates whether the handler should
+            be registered for all subclasses
 
-        This function can be also used as a decorator by omitting the `handler` argument::
+        This function can be also used as a decorator
+        by omitting the `handler` argument::
 
             @jsonpickle.handlers.register(Foo, base=True)
             class FooHandler(jsonpickle.handlers.BaseHandler):
@@ -79,7 +82,8 @@ class Registry(object):
             raise TypeError('{0!r} is not a class/type'.format(cls))
         # store both the name and the actual type for the ugly cases like
         # _sre.SRE_Pattern that cannot be loaded back directly
-        self._handlers[util.importable_name(cls)] = self._handlers[cls] = handler
+        self._handlers[util.importable_name(cls)] = \
+            self._handlers[cls] = handler
         if base:
             # only store the actual type for subclass checking
             self._base_handlers[cls] = handler
@@ -214,6 +218,7 @@ class QueueHandler(BaseHandler):
     def restore(self, data):
         return queue.Queue()
 
+
 QueueHandler.handles(queue.Queue)
 
 
@@ -228,7 +233,8 @@ class CloneFactory(object):
         return clone(self.exemplar)
 
     def __repr__(self):
-        return ('<CloneFactory object at 0x%x (%s)>' % (id(self), self.exemplar))
+        return (
+            '<CloneFactory object at 0x%x (%s)>' % (id(self), self.exemplar))
 
 
 class UUIDHandler(BaseHandler):
@@ -240,5 +246,6 @@ class UUIDHandler(BaseHandler):
 
     def restore(self, data):
         return uuid.UUID(data['hex'])
+
 
 UUIDHandler.handles(uuid.UUID)

--- a/jsonpickle/pickler.py
+++ b/jsonpickle/pickler.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2008 John Paulett (john -at- paulett.org)
-# Copyright (C) 2009, 2011, 2013 David Aguilar (davvid -at- gmail.com) and contributors
+# Copyright (C) 2009, 2011, 2013 David Aguilar (davvid -at- gmail.com)
+# and contributors
 # All rights reserved.
 #
 # This software is licensed as described in the file COPYING, which
@@ -10,12 +11,13 @@ from __future__ import absolute_import, division, unicode_literals
 import base64
 import warnings
 import sys
+import types
 from itertools import chain, islice
 
 from . import util
 from . import tags
 from . import handlers
-from .backend import JSONBackend, json
+from .backend import json
 from .compat import numeric_types, unicode, PY3, PY2
 
 
@@ -206,7 +208,7 @@ class Pickler(object):
 
     def _get_flattener(self, obj):
 
-        if PY2 and isinstance(obj, file):
+        if PY2 and isinstance(obj, types.FileType):
             return self._flatten_file
 
         if util.is_primitive(obj):
@@ -267,14 +269,14 @@ class Pickler(object):
         """
         Special case file objects
         """
-        assert not PY3 and isinstance(obj, file)
+        assert not PY3 and isinstance(obj, types.FileType)
         return None
 
     def _flatten_bytestring(self, obj):
         if PY2:
             try:
                 return obj.decode('utf-8')
-            except:
+            except Exception:
                 pass
         return {tags.B64: base64.encodestring(obj).decode('utf-8')}
 
@@ -361,8 +363,8 @@ class Pickler(object):
 
                 # check that getstate/setstate is sane
                 if not (state and hasattr(obj, '__getstate__')
-                            and not hasattr(obj, '__setstate__')
-                            and not isinstance(obj, dict)):
+                        and not hasattr(obj, '__setstate__')
+                        and not isinstance(obj, dict)):
                     # turn iterators to iterables for convenient serialization
                     if rv_as_list[3]:
                         rv_as_list[3] = tuple(rv_as_list[3])
@@ -417,7 +419,8 @@ class Pickler(object):
 
         if util.is_iterator(obj):
             # force list in python 3
-            data[tags.ITERATOR] = list(map(self._flatten, islice(obj, self._max_iter)))
+            data[tags.ITERATOR] = list(
+                map(self._flatten, islice(obj, self._max_iter)))
             return data
 
         if has_dict:
@@ -528,7 +531,7 @@ class Pickler(object):
             elif not isinstance(k, (str, unicode)):
                 try:
                     k = repr(k)
-                except:
+                except Exception:
                     k = unicode(k)
 
         data[k] = self._flatten(v)

--- a/jsonpickle/unpickler.py
+++ b/jsonpickle/unpickler.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2008 John Paulett (john -at- paulett.org)
-# Copyright (C) 2009, 2011, 2013 David Aguilar (davvid -at- gmail.com) and contributors
+# Copyright (C) 2009, 2011, 2013 David Aguilar (davvid -at- gmail.com)
+# and contributors
 # All rights reserved.
 #
 # This software is licensed as described in the file COPYING, which
@@ -15,7 +16,7 @@ from . import util
 from . import tags
 from . import handlers
 from .compat import numeric_types, set, unicode
-from .backend import JSONBackend, json
+from .backend import json
 
 
 def decode(string, backend=None, context=None, keys=False, reset=True,
@@ -198,7 +199,8 @@ class Unpickler(object):
         elif util.is_dictionary(obj):
             restore = self._restore_dict
         else:
-            restore = lambda x: x
+            def restore(x):
+                return x
         return restore(obj)
 
     def _restore_base64(self, obj):
@@ -220,7 +222,7 @@ class Unpickler(object):
         proxy = _Proxy()
         self._mkref(proxy)
         reduce_val = obj[tags.REDUCE]
-        reduce_tuple = f, args, state, listitems, dictitems = map(self._restore, reduce_val)
+        f, args, state, listitems, dictitems = map(self._restore, reduce_val)
 
         if f == tags.NEWOBJ or getattr(f, '__name__', '') == '__newobj__':
             # mandated special case
@@ -249,7 +251,7 @@ class Unpickler(object):
                     try:
                         for k, v in state.items():
                             setattr(stage1, k, v)
-                    except:
+                    except Exception:
                         dict_state, slots_state = state
                         if dict_state:
                             stage1.__dict__.update(dict_state)
@@ -344,9 +346,11 @@ class Unpickler(object):
         if kwargs:
             kwargs = self._restore(kwargs)
 
-        is_oldstyle = not (isinstance(cls, type) or getattr(cls, '__meta__', None))
+        is_oldstyle = (
+            not (isinstance(cls, type) or getattr(cls, '__meta__', None)))
         try:
-            if (not is_oldstyle) and hasattr(cls, '__new__'):  # new style classes
+            if (not is_oldstyle) and hasattr(cls, '__new__'):
+                # new style classes
                 if factory:
                     instance = cls.__new__(cls, factory, *args, **kwargs)
                     instance.default_factory = factory
@@ -363,7 +367,7 @@ class Unpickler(object):
             except TypeError:  # fail gracefully
                 try:
                     instance = make_blank_classic(cls)
-                except:  # fail gracefully
+                except Exception:  # fail gracefully
                     return self._mkref(obj)
 
         proxy.reset(instance)
@@ -476,9 +480,11 @@ class Unpickler(object):
         children = [self._restore(v) for v in obj]
         parent.extend(children)
         method = _obj_setvalue
-        proxies = [(parent, idx, value, method)
-                    for idx, value in enumerate(parent)
-                        if isinstance(value, _Proxy)]
+        proxies = [
+            (parent, idx, value, method)
+            for idx, value in enumerate(parent)
+            if isinstance(value, _Proxy)
+        ]
         self._proxies.extend(proxies)
         return parent
 
@@ -521,7 +527,8 @@ class Unpickler(object):
         if self.keys:
             restore_key = self._restore_pickled_key
         else:
-            restore_key = lambda key: key
+            def restore_key(key):
+                return key
         return restore_key
 
     def _restore_pickled_key(self, key):

--- a/jsonpickle/util.py
+++ b/jsonpickle/util.py
@@ -63,10 +63,14 @@ def has_method(obj, name):
     if not isinstance(func, (types.MethodType, types.FunctionType)):
         return False
 
-    # need to go through __dict__'s since in py3 methods are essentially descriptors
-    base_type = obj if is_type(obj) else obj.__class__  # __class__ for old-style classes
+    # need to go through __dict__'s since in py3
+    # methods are essentially descriptors
+
+    # __class__ for old-style classes
+    base_type = obj if is_type(obj) else obj.__class__
     original = None
-    for subtype in inspect.getmro(base_type):  # there is no .mro() for old-style classes
+    # there is no .mro() for old-style classes
+    for subtype in inspect.getmro(base_type):
         original = vars(subtype).get(name)
         if original is not None:
             break
@@ -342,22 +346,23 @@ def is_iterator(obj):
 def is_collections(obj):
     try:
         return type(obj).__module__ == 'collections'
-    except:
+    except Exception:
         return False
 
 
 IteratorType = type(iter(''))
 
+
 def is_reducible(obj):
     """
-    Returns false if of a type which have special casing, and should not have their
-    __reduce__ methods used
+    Returns false if of a type which have special casing,
+    and should not have their __reduce__ methods used
     """
     # defaultdicts may contain functions which we cannot serialise
     if is_collections(obj) and not isinstance(obj, collections.defaultdict):
         return True
     return (not
-                (is_list(obj) or
+            (is_list(obj) or
                 is_list_like(obj) or
                 is_primitive(obj) or
                 is_bytes(obj) or
@@ -375,7 +380,7 @@ def is_reducible(obj):
                 type(obj) is object or
                 obj is object or
                 (is_type(obj) and obj.__module__ == 'datetime')
-                ))
+             ))
 
 
 def in_dict(obj, key, default=False):
@@ -391,7 +396,9 @@ def in_slots(obj, key, default=False):
     Returns true if key exists in obj.__slots__; false if not in.
     If obj.__slots__ is absent, return default
     """
-    return (key in obj.__slots__) if getattr(obj, '__slots__', None) else default
+    return (
+        (key in obj.__slots__) if getattr(obj, '__slots__', None) else default
+    )
 
 
 def has_reduce(obj):
@@ -409,7 +416,7 @@ def has_reduce(obj):
     # notwithstanding depending on default object
     # reduce
     if is_noncomplex(obj):
-         return (False, True)
+        return (False, True)
 
     has_reduce = False
     has_reduce_ex = False
@@ -419,7 +426,7 @@ def has_reduce(obj):
 
     # For object instance
     has_reduce = in_dict(obj, REDUCE) or in_slots(obj, REDUCE)
-    has_reduce_ex = in_dict(obj, REDUCE_EX) or in_slots(obj, REDUCE_EX) 
+    has_reduce_ex = in_dict(obj, REDUCE_EX) or in_slots(obj, REDUCE_EX)
 
     # turn to the MRO
     for base in type(obj).__mro__:
@@ -429,20 +436,20 @@ def has_reduce(obj):
         if has_reduce and has_reduce_ex:
             return (has_reduce, has_reduce_ex)
 
-    # for things that don't have a proper dict but can be getattred (rare, but includes some
-    # builtins)
+    # for things that don't have a proper dict but can be
+    # getattred (rare, but includes some builtins)
     cls = type(obj)
     object_reduce = getattr(object, REDUCE)
     object_reduce_ex = getattr(object, REDUCE_EX)
     if not has_reduce:
-         has_reduce_cls = getattr(cls, REDUCE, False)
-         if not has_reduce_cls is object_reduce:
-             has_reduce = has_reduce_cls
+        has_reduce_cls = getattr(cls, REDUCE, False)
+        if has_reduce_cls is not object_reduce:
+            has_reduce = has_reduce_cls
 
     if not has_reduce_ex:
         has_reduce_ex_cls = getattr(cls, REDUCE_EX, False)
-        if not has_reduce_ex_cls is object_reduce_ex:
-             has_reduce_ex = has_reduce_ex_cls
+        if has_reduce_ex_cls is not object_reduce_ex:
+            has_reduce_ex = has_reduce_ex_cls
 
     return (has_reduce, has_reduce_ex)
 

--- a/setup.py
+++ b/setup.py
@@ -22,8 +22,10 @@ exec(open(version).read(), scope)
 SETUP_ARGS = dict(
     name='jsonpickle',
     version=scope['__version__'],
-    description='Python library for serializing any arbitrary object graph into JSON',
-    long_description='jsonpickle converts complex Python objects to and from JSON.',
+    description=(
+        'Python library for serializing any arbitrary object graph into JSON'),
+    long_description=(
+        'jsonpickle converts complex Python objects to and from JSON.'),
     author='David Aguilar',
     author_email='davvid@gmail.com',
     url='http://jsonpickle.github.io/',

--- a/tests/backend_test.py
+++ b/tests/backend_test.py
@@ -138,7 +138,7 @@ class JsonlibTestCase(BackendBase):
             return self.skip('no jsonlib for python3')
         expected_pickled = (
             '{"things":[{'
-            '"py\/object":"backend_test.Thing",'
+            r'"py\/object":"backend_test.Thing",'
             '"name":"data","child":null}'
             ']}')
         self.assertEncodeDecode(expected_pickled)
@@ -165,7 +165,7 @@ class UJsonTestCase(BackendBase):
     def test_backend(self):
         expected_pickled = (
             '{"things":[{'
-            '"py\/object":"backend_test.Thing",'
+            r'"py\/object":"backend_test.Thing",'
             '"name":"data","child":null}'
             ']}')
         self.assertEncodeDecode(expected_pickled)

--- a/tests/benchmark.py
+++ b/tests/benchmark.py
@@ -32,6 +32,6 @@ if doc['feed']['title'] != unpickled['feed']['title']:
     print 'Not a match'
 """ % mod
 
-print 'Using %s' % mod
+print('Using %s' % mod)
 json_test = timeit.Timer(stmt=json)
-print "%.9f sec/pass " % (json_test.timeit(number=number) / number)
+print("%.9f sec/pass " % (json_test.timeit(number=number) / number))

--- a/tests/datetime_test.py
+++ b/tests/datetime_test.py
@@ -34,6 +34,7 @@ class UTC(datetime.tzinfo):
     def dst(self, dt):
         return datetime.timedelta()
 
+
 utc = UTC()
 
 
@@ -258,6 +259,7 @@ def suite():
     suite.addTest(unittest.makeSuite(DateTimeAdvancedTestCase))
     suite.addTest(unittest.makeSuite(DateTimeInnerReferenceTestCase))
     return suite
+
 
 if __name__ == '__main__':
     unittest.main(defaultTest='suite')

--- a/tests/document_test.py
+++ b/tests/document_test.py
@@ -99,5 +99,6 @@ def suite():
     suite.addTest(unittest.makeSuite(DocumentTestCase))
     return suite
 
+
 if __name__ == '__main__':
     unittest.main(defaultTest='suite')

--- a/tests/handler_test.py
+++ b/tests/handler_test.py
@@ -132,10 +132,10 @@ class HandlerTestCase(unittest.TestCase):
 
     def test_custom_handler_can_rewrite_everything(self):
         """Test the low-level pickling structures"""
-        jsonpickle.unregister(CustomObject)
-        jsonpickle.register(CustomObject, SithHandler)
+        jsonpickle.handlers.unregister(CustomObject)
+        jsonpickle.handlers.register(CustomObject, SithHandler)
         obj = CustomObject('jarjar')  # secret sith lord jarjar
-        pickler = jsonpickle.Pickler()
+        pickler = jsonpickle.pickler.Pickler()
         data = pickler.flatten(obj)
         self.assertTrue(isinstance(data, dict))
         self.assertEqual(len(data), 2)

--- a/tests/handler_test.py
+++ b/tests/handler_test.py
@@ -54,7 +54,6 @@ class DecoratedHandler(NullHandler):
     pass
 
 
-
 class SithHandler(jsonpickle.handlers.BaseHandler):
     """(evil) serialization handler that rewrites the entire object"""
 

--- a/tests/jsonpickle_test.py
+++ b/tests/jsonpickle_test.py
@@ -1375,5 +1375,6 @@ def suite():
     suite.addTest(doctest.DocTestSuite(jsonpickle.unpickler))
     return suite
 
+
 if __name__ == '__main__':
     unittest.main(defaultTest='suite')

--- a/tests/jsonpickle_test.py
+++ b/tests/jsonpickle_test.py
@@ -399,6 +399,22 @@ class JSONPickleTestCase(SkippableTest):
             % tags.OBJECT
         )
 
+    def test_API_names(self):
+        """
+        Enforce expected names in main module
+        """
+        names = list(vars(jsonpickle))
+        self.assertIn('pickler', names)
+        self.assertIn('unpickler', names)
+        self.assertIn('JSONBackend', names)
+        self.assertIn('__version__', names)
+        self.assertIn('register', names)
+        self.assertIn('unregister', names)
+        self.assertIn('Pickler', names)
+        self.assertIn('Unpickler', names)
+        self.assertIn('encode', names)
+        self.assertIn('decode', names)
+
     def test_encode(self):
         expect = self.obj
         pickle = jsonpickle.encode(self.obj)

--- a/tests/numpy_test.py
+++ b/tests/numpy_test.py
@@ -113,7 +113,8 @@ class NumpyTestCase(SkippableTest):
 
         if not PY2:
             arrays.extend([
-                np.rec.array([('NGC1001', 11), ('NGC1002', 1.), ('NGC1003', 1.)],
+                np.rec.array([
+                    ('NGC1001', 11), ('NGC1002', 1.), ('NGC1003', 1.)],
                              dtype=[('target', 'S20'), ('V_mag', 'f4')])
             ])
         for array in arrays:

--- a/tests/object_test.py
+++ b/tests/object_test.py
@@ -744,7 +744,9 @@ class AdvancedObjectsTestCase(SkippableTest):
 
         restore = self.unpickler.restore
         flatten = self.pickler.flatten
-        roundtrip = lambda obj: restore(flatten(obj))
+
+        def roundtrip(obj):
+            return restore(flatten(obj))
         self.assertTrue(roundtrip(IntEnumTest.X) is IntEnumTest.X)
         self.assertTrue(roundtrip(IntEnumTest) is IntEnumTest)
 
@@ -792,7 +794,7 @@ class AdvancedObjectsTestCase(SkippableTest):
             self.assertEqual(type(encoded), unicode)
         else:
             self.assertNotEqual(encoded, u1)
-            b64ustr= base64.encodestring(b'foo').decode('utf-8')
+            b64ustr = base64.encodestring(b'foo').decode('utf-8')
             self.assertEqual({tags.B64: b64ustr}, encoded)
             self.assertEqual(type(encoded[tags.B64]), unicode)
         decoded = self.unpickler.restore(encoded)
@@ -805,7 +807,7 @@ class AdvancedObjectsTestCase(SkippableTest):
         # bytestrings that we can't decode to UTF-8 will always be wrapped
         encoded = self.pickler.flatten(b2)
         self.assertNotEqual(encoded, b2)
-        b64ustr= base64.encodestring(b'foo\xff').decode('utf-8')
+        b64ustr = base64.encodestring(b'foo\xff').decode('utf-8')
         self.assertEqual({tags.B64: b64ustr}, encoded)
         self.assertEqual(type(encoded[tags.B64]), unicode)
         decoded = self.unpickler.restore(encoded)
@@ -854,6 +856,7 @@ class UnicodeMixinHandler(handlers.BaseHandler):
 
     def restore(self, obj):
         return UnicodeMixin(obj['value'])
+
 
 handlers.register(UnicodeMixin, UnicodeMixinHandler)
 

--- a/tests/runtests.py
+++ b/tests/runtests.py
@@ -10,8 +10,8 @@
 import os
 import sys
 
-testdir = os.path.dirname(os.path.abspath(__file__))
-sys.path.insert(1, os.path.dirname(testdir))
+testdir = os.path.dirname(os.path.abspath(__file__))  # noqa: E402
+sys.path.insert(1, os.path.dirname(testdir))  # noqa: E402
 
 import unittest
 

--- a/tests/sqlalchemy_test.py
+++ b/tests/sqlalchemy_test.py
@@ -94,9 +94,10 @@ class SQLAlchemyTestCase(SkippableTest):
             return self.skip('sqlalchemy is not installed')
 
         meta = sqa.MetaData()
-        expect = sqa.Table('test', meta,
-                sqa.Column('id', sqa.Integer()),
-                sqa.Column('text', sqa.Text()))
+        expect = sqa.Table(
+            'test', meta,
+            sqa.Column('id', sqa.Integer()),
+            sqa.Column('text', sqa.Text()))
 
         jsonstr = jsonpickle.dumps(expect)
         actual = jsonpickle.loads(jsonstr)
@@ -115,6 +116,7 @@ class SQLAlchemyTestCase(SkippableTest):
         self.assertEqual(expect.c.text.name, actual.c.text.name)
         self.assertEqual(expect.c.text.type.__class__,
                          actual.c.text.type.__class__)
+
 
 def suite():
     suite = unittest.TestSuite()

--- a/tests/util_test.py
+++ b/tests/util_test.py
@@ -238,5 +238,6 @@ def suite():
     suite.addTest(doctest.DocTestSuite(jsonpickle.util))
     return suite
 
+
 if __name__ == '__main__':
     unittest.main(defaultTest='suite')

